docs/topics/downloader-middleware.rst

```diff
@@ -729,6 +729,13 @@ RedirectMiddleware
 The urls which the request goes through (while being redirected) can be found
 in the ``redirect_urls`` :attr:`Request.meta <scrapy.http.Request.meta>` key.
 
+.. reqmeta:: redirect_reasons
+
+The status of redirected urls can be found in the ``redirect_reasons``
+:attr:`Request.meta <scrapy.http.Request.meta>` key. This list can have
+either ``int`` HTTP status codes (Ex:- 'redirect_reasons': [301, 302, 402])
+or ``string`` 'meta refresh' (Ex:- 'redirect_reasons': ['meta refresh'])
+
```

Comment:
I still see some issues: the description starts saying that the list contains the _status_ of redirected URLs, and the examples may seem like you get either status codes or the 'meta refresh' string, while both can appear on the same list.

What about this?

```
The reason behind each redirect in :reqmeta:`redirect_urls` can be found in the
``redirect_reasons`` :attr:`Request.meta <scrapy.http.Request.meta>` key. For
example: ``[301, 301, 302, 'meta refresh']``.

The format of a reason depends on the middleware that handled the corresponding
redirect. For example, :class:`RedirectMiddleware` indicates the triggering
response status code as an integer, while :class:`MetaRefreshMiddleware`
always uses the ``'meta refresh'`` string as reason.
```

Reply:
Writing docs was much difficult than fixing bug haha.
Thanks for helping @Gallaecio :)